src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -301,82 +320,128 @@ func (f *NAPConfig) getMetadata() (*models.Metadata, error) {
-					break
-				}
+			if v.ParameterData != (ParameterData{}) {
+				isB64Decoded = v.ParameterData.IsBase64Decoded
+				name = v.ParameterData.Name
+				value = v.ParameterData.Value
+			} else if v.ParamData != (ParamData{}) {
+				isB64Decoded = v.ParamData.IsBase64Decoded
+				name = v.ParamData.Name
+				value = v.ParamData.Value
+			} else if v.ParamName != "" {
+				isB64Decoded = v.IsBase64Decoded
+				name = v.ParamName
+			} else {
+				logger.Warn("context is parameter but no Parameter data received")
+			}
 
+			if isB64Decoded {
 				violation.ContextData = &models.ContextData{
-					Name:  string(decodedName),
-					Value: string(decodedValue),
-				}
-			} else if v.ParamData != (ParamData{}) {
-				decodedName, err := base64.StdEncoding.DecodeString(v.ParamData.Name)
-				if err != nil {
-					logger.Errorf("could not decode the Paramater Name %s for %v", v.ParamData.Name, f.SupportID)
-					break
+					Name:  name,
+					Value: value,
 				}
+				break
+			}
 
-				decodedValue, err := base64.StdEncoding.DecodeString(v.ParamData.Value)
-				if err != nil {
-					logger.Errorf("could not decode the Paramater Value %s for %v", v.ParamData.Value, f.SupportID)
-					break
-				}
+			decodedName, err := base64.StdEncoding.DecodeString(name)
+			if err != nil {
+				logger.Errorf("could not decode the Paramater Name %s for %v", name, f.SupportID)
+				break
+			}
 
-				violation.ContextData = &models.ContextData{
-					Name:  string(decodedName),
-					Value: string(decodedValue),
-				}
-			} else if v.Context == parameterCtx {
-				logger.Warn("context is parameter but no Parameter data received")
+			decodedValue, err := base64.StdEncoding.DecodeString(value)
+			if err != nil {
+				logger.Errorf("could not decode the Paramater Value %s for %v", value, f.SupportID)
+				break
+			}
+
+			violation.ContextData = &models.ContextData{
+				Name:  string(decodedName),
+				Value: string(decodedValue),
 			}
 		case headerCtx:
-			if v.Header == (Header{}) {
-				logger.Warn("context is header but no Header data received")
+			var isB64Decoded bool
+			var name, value string
+
+			if v.Header != (Header{}) {
+				isB64Decoded = v.Header.IsBase64Decoded
+				name = v.Header.Name
+				value = v.Header.Value
+			} else if v.HeaderData != (Header{}) {
+				isB64Decoded = v.HeaderData.IsBase64Decoded
+				name = v.HeaderData.Name
+				value = v.HeaderData.Value
+			}
+
+			if isB64Decoded {
+				violation.ContextData = &models.ContextData{
+					Name:  name,
+					Value: value,
+				}
 				break
 			}
 
-			decodedName, err := base64.StdEncoding.DecodeString(v.Header.Name)
+			decodedName, err := base64.StdEncoding.DecodeString(name)
 			if err != nil {
-				logger.Errorf("could not decode the Header Name %s for %v", v.Header.Name, f.SupportID)
+				logger.Errorf("could not decode the Header Name %s for %v", name, f.SupportID)
 				break
```

Comment:
I know that this wasn't added in this PR, but are we sure that we want to `break` if we fail to base64 decode? I would think that we'd want to display the name and value even if there's a decoding issue, rather than returning an empty context?

Reply:
I don't think it's useful to display Base64 data that failed to be decoded, what do you think? It might create more confusion. The user might not even be aware that it's b64 encoded.

Reply:
Yeah I was thinking that there could be a case where it's got some other encoding that we haven't accounted for? But perhaps it's not a good a user experience to get encoded values in the front end 🤔